plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1066 @@
+
+        if version1 != version2:
+            raise ValueError("incompatible IP versions")
+        return ip_lower <= ip_integer <= ip_upper
+
+    @staticmethod
+    def ip_to_integer(ip_address):
+        """Converts an IP address
+
+        Converts an IP address expressed as a string to its representation as an integer value and returns a tuple
+        (ip_integer, version), with version being the IP version (either 4 or 6).
+
+        Args:
+            str(ip_address): IP-Address e.g. 192.168.178.99 or fd42:bd00:7b11:2167:216:3eff:fe21:e86b
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            tuple(
+                int(ip_integer): int ip representation
+                int(ip version): 4 for IPv4 or 6 for IPv6
+                )"""
+        # try parsing the IP address first as IPv4, then as IPv6
+        for version in (socket.AF_INET, socket.AF_INET6):
+            try:
+                ip_hex = socket.inet_pton(version, ip_address)
+                ip_integer = int(binascii.hexlify(ip_hex), 16)
+                return (ip_integer, 4 if version == socket.AF_INET else 6)
+            except Exception:
+                pass
+        raise ValueError("invalid IP address")
+
+    @staticmethod
+    def subnetwork_to_ip_range(subnetwork):
+        """Returns the uper and lower ip representations of the IP-Subnet.
+
+        Returns a tuple (ip_lower, ip_upper, version) containing the integer values of the lower and upper IP addresses respectively
+        in a subnetwork expressed in CIDR notation (as a string), with version being the subnetwork IP version (either 4 or 6).
+
+        Args:
+            str(subnetwork): IP-Subnet e.g. 192.168.178.0/24 or fd42:bd00:7b11:2167:216:3eff::/44
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            tuple(
+                int(ip_lower): int ip representation lower subnet end
+                int(ip_upper): int ip representation upper subnet end
+                int(ip version): 4 for IPv4 or 6 for IPv6
+                )"""
+        try:
+            fragments = subnetwork.split('/')
+            network_prefix = fragments[0]
+            netmask_len = int(fragments[1])
+
+            # try parsing the subnetwork first as IPv4, then as IPv6
+            for version in (socket.AF_INET, socket.AF_INET6):
+                ip_len = 32 if version == socket.AF_INET else 128
+                try:
+                    suffix_mask = (1 << (ip_len - netmask_len)) - 1
+                    netmask = ((1 << ip_len) - 1) - suffix_mask
+                    ip_hex = socket.inet_pton(version, network_prefix)
+                    ip_lower = int(binascii.hexlify(ip_hex), 16) & netmask
+                    ip_upper = ip_lower + suffix_mask
+                    return (ip_lower,
+                            ip_upper,
+                            4 if version == socket.AF_INET else 6)
+                except Exception:
```

Comment:
Wouldn't it be better to use [ipaddress](https://docs.python.org/3/library/ipaddress.html) instead of doing such computations manually?

Reply:
I thought about that too, but apparently `ipaddress` only came around in Python 3.3

Reply:
It can be installed manually for older Python versions.

Reply:
https://pypi.org/project/ipaddress/

Reply:
As russoz explained it is included by Python v. 3.x and i will rewrite this if the ansible support of python 2.x is dropped. My goal is that no further libraries have to be installed. This makes it even more complicated from the user view if he had to install other librarys.

Reply:
I searched ansible repo itself for that and I found
```python
from ansible.module_utils.compat.ipaddress import ip_network, IPv6Network
```
but apparently that comes from `ansible.netcommon.compat.ipaddress` (there is a redirect).

@felixfontein I recall we were getting rid of these collection dependencies, but not sure where we are with that particular one.

Reply:
I remember that because of this we vendored it some time ago: just use `from ansible_collections.community.general.plugins.module_utils.compat import ipaddress`.

(But even if we would not have: require users to install this library manually for old Pythons is totally acceptable.)

Reply:
I still believe that importing/installing `ipaddress` would make this plugin's code leaner, but I can live with that being done in a separate PR.

Reply:
Fixed in d6cc38b9ad31b7fb291b9a823141e2f4e1489052.